analyze-project/action.yml

```
@@ -0,0 +1,94 @@
+name: Analyze Python Project
+description: >
+  This workflow checks the code quality of a Python project using various
+  analyzers like linters and type checkers, including ni-python-styleguide,
+  mypy, bandit, and pyright. It is designed to be reusable across different
+  projects and can be easily integrated into existing CI/CD pipelines.
+
+inputs:
+  project-directory:
+    description: Path to the directory containing pyproject.toml.
+    default: ${{ github.workspace }}
+  extras:
+    # E.g. "docs drivers"
+    description: 'List of Poetry extras to install (separated by spaces)'
+    default: ''
+    required: false
+    type: string
+
+runs:
+  using: composite
+  steps:
+    - name: Get package name and version
+      id: get_package_info
+      run: |
+        result=$(poetry version -C "${{ inputs.project-directory }}")
+        name=$(echo "$result" | awk '{print $1}')
+        version=$(echo "$result" | awk '{print $2}')
+        echo "name=$name" >> $GITHUB_OUTPUT
+        echo "version=$version" >> $GITHUB_OUTPUT
+      shell: bash
+    - name: Check for lock changes
+      run: poetry check --lock -C "${{ inputs.project-directory }}"
+      shell: bash
+    - name: Cache virtualenv
+      uses: actions/cache@0400d5f644dc74513175e3cd8d07132dd4860809 # v4.2.4
+      with:
+        path: ${{ inputs.project-directory }}/.venv
+        key: ${{ inputs.package-name }}-${{ runner.os }}-py${{ steps.setup-python.outputs.python-version }}-${{ hashFiles(format('{0}/poetry.lock', inputs.project-directory)) }}
+    - name: Install ${{ steps.get_package_info.outputs.name }}
+      run: |
+        if [ "${{ inputs.extras }}" != "" ]; then
+          EXTRAS_ARGS=""
+          for extra in ${{ inputs.extras }}; do
+            EXTRAS_ARGS="$EXTRAS_ARGS -E $extra"
+          done
+          poetry install -v $EXTRAS_ARGS
+        else
+          poetry install -v
+        fi
```

Poetry accepts space-delimited extras. Remember https://github.com/ni/ni-apis-python/blob/main/.github/workflows/check_analyzers.yml#L47 ?

```suggestion
        if [ -n "${{ inputs.extras }}" ]; then
          poetry install -v --extras '${{ inputs.extras }}'
        else
          poetry install -v
        fi
```